### PR TITLE
Use Directory.Build.props/targets

### PR DESCRIPTION
**Pull request by natemcmaster**
Adopts a pattern we've started using in Kestrel, Mvc, and other repos. It helps us reduce duplication between csproj files.

Also, I moved the microbenchmarks project out of test/ into benchmarks/ because it isn't actually a test project.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -1,20 +1,16 @@
...
+﻿<Project>
+  <Import Project="build\dependencies.props" />
+  <Import Project="version.xml" />
 
   <PropertyGroup>
     <Product>Microsoft ASP.NET Core</Product>
     <RepositoryUrl>https://github.com/aspnet/SignalR</RepositoryUrl>
     <RepositoryType>git</RepositoryType>
-    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)Key.snk</AssemblyOriginatorKeyFile>
+    <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)build\Key.snk</AssemblyOriginatorKeyFile>
```
Comment (moozzyk) [inline Directory.Build.props]: does it mean we now have a build/build folder?

File: build/repo.props [hunk 2]
```diff
@@ -1,10 +1,5 @@
 <Project>
   <ItemGroup>
-    <ExcludeFromTest Include="$(RepositoryRoot)test\Microsoft.AspNetCore.SignalR.Test.Server\*.csproj" />
```
Comment (moozzyk) [inline build/repo.props]: why is this no longer needed? Are tests project identified differently now?

**Comment by davidfowl** [comment]
Do we want to do this before the alpha?

File: Directory.Build.props (hunk 1 above)
Comment (natemcmaster) [reply Directory.Build.props]: No. This file moved up a level into the repo root.

File: build/repo.props (hunk 2 above)
Comment (natemcmaster) [reply build/repo.props]: These project files no longer exist in these locations. Because they moved, this exclusion isn't necessary.

**Comment by natemcmaster** [comment]
> Do we want to do this before the alpha?

The impact of this change is minimal. PackageLineup, on the other hand, deserves more discussion. ==> #791

**Comment by moozzyk** [comment]
LGTM

**Comment by natemcmaster** [comment]
Ok. I'll put this in for now. Fortunately, nothing in VS, KoreBuild, or TC requires us to use this, so we can revert if we change our minds later.

**Pull request merged**